### PR TITLE
msms recipe

### PR DESCRIPTION
**Pull request by rhs2132**
This is a simple recipe for installing msms. A similar package exists on bioconda but intentionally excludes windows compatibility. Please let me know if I need to make any changes.

Ryan

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by rhs2132** [comment]
Thanks in advance for the help getting this to work. I have been trying to test the recipe locally as recommended using the .circleci/run_docker_build.sh for linux64. Everything runs seemingly correctly but exists with the message "Found no recipes to build". I think that I am missing something obvious but can't figure out what it is. I have attached the stdio output from the docker script.


[docker_log.txt](https://github.com/conda-forge/staged-recipes/files/4743308/docker_log.txt)

**Comment by rhs2132** [comment]
I fixed the previous issue: my repository was pointing to my fork so my recipe was getting automatically removed from the recipe folder. I added the build requirement of the c compiler ({{ compiler('c') }}) to the recipe which helped a little. I am now getting an overlinking error. I have been searching the forums but have not found a fix yet.
[docker_log.txt](https://github.com/conda-forge/staged-recipes/files/4747348/docker_log.txt)
 I have pasted the specific warnings/errors below with the full stdio output attached:

WARNING :: get_rpaths_raw()=[] and patchelf=[['']] disagree for /home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/msms :: 
WARNING: Disagreement in get_linkages(filename=/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/msms, resolve_filenames=True, recurse=False, sysroot=/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_build_env/x86_64-conda-linux-gnu/sysroot/, envroot=/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl, arch=native):
 lief: {'/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_build_env/x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6', '/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_build_env/x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6'}
pyldd: {'/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_build_env/x86_64-conda-linux-gnu/sysroot/lib/libc.so.6', '/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_build_env/x86_64-conda-linux-gnu/sysroot/lib/libm.so.6'}
  (using lief)
  ERROR (msms-exe,bin/msms): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 not found in any CDT/compiler package, nor the whitelist?!
  ERROR (msms-exe,bin/msms): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 not found in any CDT/compiler package, nor the whitelist?!
Traceback (most recent call last):
  File "/home/conda/.ci_support/build_all.py", line 134, in <module>
    build_all(args.recipes_dir, args.arch)
  File "/home/conda/.ci_support/build_all.py", line 67, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File "/home/conda/.ci_support/build_all.py", line 124, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2864, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2151, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1505, in bundle_conda
    files = post_process_files(metadata, initial_files)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1364, in post_process_files
    post_build(m, new_files, build_python=python)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1178, in post_build
    check_overlinking(m, files, host_prefix)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1110, in check_overlinking
    m.config.enable_static)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1072, in check_overlinking_impl
    raise OverLinkingError(overlinking_errors)
conda_build.exceptions.OverLinkingError: overlinking check failed 
['  ERROR (msms-exe,bin/msms): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (msms-exe,bin/msms): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 not found in any CDT/compiler package, nor the whitelist?!']

**Comment by rhs2132** [comment]
@conda-forge/staged-recipes This is ready for review/merge. This is modeled after the bioconda msms package but with added windows compatibility. The license is custom. I pulled the relevant file into the packaged LICENSE.txt and classified it as AFL-3.0 after comparing the license with the SPDX options. Let me know if I need to make any more changes. Thanks!

**Comment by scopatz** [comment]
> The name change to bio-msms is to avoid confusion with the bioconda package

We should probably have consistent naming

**Comment by rhs2132** [comment]
I changed the name to msms. You're right that if the packages are functionally identical on systems where they are both available that they should have the same name.

@conda-forge/bioconda-recipes What do you think the proper name should be? I have been using the bioconda msms package but want to deploy some code with the same dependency onto windows. I have not replicated the pdb_to_xyzr and pdb_to_xyzrn scripts in this package because identical functionality is available through biopython albeit through a private method. I can reincorporate this functionality for unix pretty easily so that it is a direct, drop-in replacement for the bioconda package on unix systems.

File: recipes/bio-pyvol/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    - biopython >=1.73
+    - configparser
+    - msms >=2.6.1
+    - numpy >=1.16.1
+    - pandas >=0.24.1
+    - pip
+    - python
+    - scikit-learn >=0.20.2
+    - scipy >=1.2.1
+    - trimesh >=2.36.29
```
Comment (scopatz) [inline recipes/bio-pyvol/meta.yaml:32]: Are most of these needed for building the package?

File: recipes/msms/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  name: msms
+  version: {{ version }}
+
+source:
+  - url: {{ url_base }}/REL{{ version }}/msms_i86_64Linux2_{{ version }}.tar.gz  # [linux]
+    md5: 650d8c6fd49d9ce291f30ed530d5f313  # [linux]
+  - url: {{ url_base }}/REL{{ version }}/msms_MacOSX_{{ version }}.tar.gz  # [osx]
+    md5: 36bd8447908d12a8703f458564cbe74b  # [osx]
+  - url: {{ url_base }}/REL{{ version }}/msms_win32_{{ version }}.zip  # [win]
+    md5: 482c38ca00a2c091cdc0807a8a272386  # [win32]
```
Comment (scopatz) [inline recipes/msms/meta.yaml:14]: Please build from source

Comment (rhs2132) [reply recipes/msms/meta.yaml:14]: What does that mean exactly? Scrips has only released this code as precompiled executables. I have modeled my recipe after the one in bioconda: 
https://github.com/bioconda/bioconda-recipes/blob/master/recipes/msms/meta.yaml

File: recipes/bio-pyvol/meta.yaml (hunk 1 above)
Comment (rhs2132) [reply recipes/bio-pyvol/meta.yaml:32]: I just did a more thorough re-read of the descriptions for the build, host, and run sections. I'm cleaning this up.

File: recipes/msms/meta.yaml (hunk 2 above)
Comment (scopatz) [reply recipes/msms/meta.yaml:14]: probably worth asking them

Comment (rhs2132) [reply recipes/msms/meta.yaml:14]: I have reached out to the authors and maintainers at Scripps (through MGL Tools) several times over the last year, but I have yet to hear back. I have also checked distribution for tools that rely on this executable. CCP4, VMD, Biopython, and PyMOL all can check for the presence of MSMS and read its output but do not distribute the executable itself. I would be comfortable using the Bioconda channel's version except that it excludes Windows compatibility.

What steps can I take to move this forward?

Comment (scopatz) [reply recipes/msms/meta.yaml:14]: Yeah this seems to be the only way then.  Can you please use sha256 for your hashes here, though.  I think the various bots might break if you use md5

**Pull request merged**